src/decode_r2007.c

```diff
@@ -609,7 +609,7 @@ read_system_page(Bit_Chain* dat, int64_t size_comp, int64_t size_uncomp,
                 (long)page_size, dat->size - dat->byte);
       return NULL;
     }
-  data = (BITCODE_RC*)calloc(size_uncomp, page_size);
+  data = (BITCODE_RC*)calloc(size_uncomp + page_size, 1);
```

Comment:
shouldn't this be *?

Reply:
It look like 2 arrays. `data` and `rsdata `
```
rsdata = &data[size_uncomp];
bit_read_fixed(dat, rsdata, page_size);
//...
decompress_r2007(data, size_uncomp, pedata, size_comp);
```